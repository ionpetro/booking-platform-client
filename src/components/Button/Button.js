const sizes = {
  small: 'small',
  regular: 'regular',
  large: 'large'
};

const variants = {
  light: 'light',
  dark: 'dark',
  success: 'success',
  danger: 'danger',
  primary: 'primary',
  secondary: 'secondary'
};
export default {
  props: {
    size: {
      type: String,
      default: sizes.regular,
      validator: value => Object.values(sizes).includes(value)
    },
    variant: {
      type: [String, null],
      default: variants.primary,
      validator: value => Object.values(variants).includes(value)
    }
  }
};
