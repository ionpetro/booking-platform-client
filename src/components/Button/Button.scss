@mixin button($color) {
  background: $color;

  &:hover {
    @if $color == $primary {
      background-color: $orange-dim;
    } @else {
      background-color: $blue-dim;
    }
  }
  &:disabled {
    @if $color == $primary {
      background-color: $orange-light;
    } @else {
      background-color: $blue-light;
    }
  }
  &:focus-visible {
    @if $color == $primary {
      box-shadow: 0 0 0 2px $orange-light;
    } @else {
      box-shadow: 0 0 0 2px $blue-light;
    }
  }

  // The button size is relevant to the window size
  @media #{$tablet} {
    padding: rem(12) rem(36);
  }

  // The button size is relevant to the window size
  @media #{$mobile} {
    padding: rem(8) rem(24);
  }
}

.button {
  color: white;
  font-weight: 500;
  padding: rem(16) rem(42);
  outline: none;
  text-decoration: none;
  border-radius: 10px;
  font-size: $base-font-size;
  border: none;
  display: flex;
}

.primary {
  @include button($primary);
}

.secondary {
  @include button($secondary);
}
