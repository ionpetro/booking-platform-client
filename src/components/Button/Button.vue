<template>
  <button class="btn" :class="[variant, size]">
    <slot>Button</slot>
  </button>
</template>

<script src="./Button.js"></script>

<style src="./Button.scss" lang="scss"></style>
