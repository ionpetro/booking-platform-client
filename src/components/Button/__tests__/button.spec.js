import {mount} from '@vue/test-utils';
import Button from '../Button.vue';

describe('Button', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(Button);
  });

  describe('when loaded', () => {
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
