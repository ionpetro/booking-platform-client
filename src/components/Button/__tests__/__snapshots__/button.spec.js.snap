// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button when loaded renders correctly 1`] = `
<button
  class="button primary"
>
  Button
</button>
`;
